### ISSUE TITLE
license question

### ISSUE THREAD

**Issue by IzzySoft** [depler/transmission-android#3]
Can you please add a license to your repo here? I guess you just forgot to declare it. As most of the libraries used are (dual) licensed GPL-2.0-only or GPL-3.0-only, I assume the same also applies to your app?

**Comment by depler** [comment]
I' am not sure why and how ) Feel free to create PR to add license 😊

**Comment by IzzySoft** [comment]
Would be a requirement for #4 anyway. OK, so you want me to pick one for you? Let me give you a few choices then to pick from those I at least have a rough idea of. For reference, see [SPDX](https://spdx.org/licenses/), we'll need one that is approved by OSI/FSF as being free/libre:

* Unlicense: best corresponds to "public domain" (approved by both, OSI+FSF), alternatively WTFPL does as well (FSF only)
* Apache-2, MIT, BSD have few conditions. And depending on which, BSD is quite short and easy to comprehend.
* GPL enforces that derivates must be GPL as well if you plan to publish derivates
* AGPL additionally enforces to open your code changes (under AGPL of course) if you run the resulting code for "public services"

There are plenty of others, feel free to dig in. But if you do not care to much about the dots on the ï, just pick one of those mentioned that suits you best.

Oh wait, stupid me: you use GPL'd libraries – so see, now your app is bound to that as GPL enforces it (section 5). So you basically just have to check whether it should be GPL-2.0 or GPL-3.0 (as the libs are dual-licensed if I got that right). IIRC, they said "only" in both cases (otherwise you'd have the option of saying "GPL-2.0-or-later").

**TL;DR:** Which of the 2 you want to pick (GPL-2.0 or GPL-3.0), or do you want to dual-license, too?

So how to apply it? Simply add the corresponding `LICENSE.txt` to the root of your repo (for GPL-3.0, e.g. copy it [from here](https://www.gnu.org/licenses/gpl-3.0.txt) and name it `LICENSE.txt`). Optionally add a header to all your code files (as the libs do, see e.g. libtransmission for an example – though I'd skip the last two lines for your files as you're not bound to the company mentioned) and a note to the Readme.

**Comment by IzzySoft** [comment]
@depler any preference yet? As shown above, it will need to be a GPL one if you're using GPL'd code. Let's see:

* b64 is public domain, so no restrictions apply.
* curl uses the curl license, which is not approved by either OSI or FSF (not sure why, the terms sound pretty free/libre to me) – so not sure about implications. Assuming none in this context.
* daemon says either GPL-2.0-only or GPL-3.0-only
* dht seems to be MIT
* event2 looks like BSD-3-Clause
* [fastfloat](https://github.com/fastfloat/fast_float) is partly Apache-2.0 and partly MIT (dual-licensed)
* [fmt](https://github.com/fmtlib/fmt) is MIT
* jsonsl is MIT
* libdeflate is MIT
* libspl is MIT
* libtransmission again is dual-licensed GPL-2.0-only / GPL-3.0-only
* libutp is MIT
* math uses the [Boost](https://spdx.org/licenses/BSL-1.0.html) license, which looks quite similar to MIT
* [MiniUPnP](https://github.com/miniupnp/miniupnp) is BSD-3-Clause
* natpmp is BSD-3-Clause
* [utfcpp](https://github.com/nemtrif/utfcpp) is Boost again
* [wildmat](https://github.com/richsalz/wildmat) is public domain
* wolfssl is GPL-2.0-or-later
* zlib most likely uses the [zlib License](https://spdx.org/licenses/Zlib.html)

So this leaves you the choice between either GPL-2.0-only, GPL-3.0-only or a dual-licensing with both (as GPL is the strongest one here it fulfills all the terms of the others but adds on top, which is why you cannot pick one of the "less-strict" licenses). IANAL, but that's how I "was taught" about it.

So will you choose?

**Comment by depler** [comment]
@IzzySoft I don't really understand what problem this license file will solve. But again, feel free to create PR and I will merge it. Let's choose GPL-3 if it matters.

**Comment by IzzySoft** [comment]
Thanks @depler – PR opened. And it solves the problem of missing clarity. Quoting [Wikipedia](https://en.wikipedia.org/wiki/License):

> A license (or licence) is an official permission or permit to do, use, or own something (as well as the document of that permission or permit).

In terms of free/libre software it ensures the freedom is granted (e.g. to use, distribute, modify etc. pp.).

I've added the `closes` clause to the PR – so when that's merged it should automatically close this issue.

Thanks a lot! Once it's merged I will see to include your app ASAP with [my repo](https://apt.izzysoft.de/fdroid), which often serves as stepping stone for F-Droid.org – and #4 can be proceeded as well. Maybe @pentaphon wants to start the process then by opening an [RFP](https://gitlab.com/fdroid/rfp/-/issues/)?

**Comment by IzzySoft** [comment]
Thanks a lot! As promised I've set everything up for my repo now, so Transmission will become [available here](https://apt.izzysoft.de/fdroid/index/apk/com.depler.transmission) with the next sync around 6 pm UTC. Be welcome to [pick a badge](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it) if you want to link there e.g. from the Readme then :smiley:

**Comment by depler** [comment]
@IzzySoft thanks for including app into your repo! Please fix changelog url, it has misprint `https://github.com/depler/transmission-android/**releasses**`

**Comment by IzzySoft** [comment]
Argh… Thanks for spotting that! Fixed, effective with the next sync.
